Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -309,6 +308,8 @@ bool MediaPlayerPrivateGStreamerMSE::doSeek(const MediaTime& position, float rat
                 loadingFailed(MediaPlayer::NetworkState::Empty);
             else
                 m_isSeeking = true;
+        } else if(state < GST_STATE_PAUSED && !prevSeekPending) {
```

Comment:
app may trigger seek multiple times, and there is no guarantee that the app will provide data for the pending one (even though in the practice they do).  this is probably a different issue.

Reply:
Will it make sense to force it on `if(state < GST_STATE_PAUSED && (!prevSeekPending || m_mediaSource->isSeeking()))`

Reply:
The isSeeking() is not present right now in the MediaSourcePrivateClient...

Reply:
I suspect that it wouldn't reach doSeek if seek is pending, see MediaPlayerPrivateGStreamerMSE::seek
```    if (m_isSeeking && m_isSeekPending) {
        m_seekTime = time;
        return;
    }
```